src/test/ui/mismatched_types/issue-38371.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-38371.rs:14:8
+   |
+14 | fn foo(&foo: Foo) {  // illegal syntax
+   |        ^^^^ expected struct `Foo`, found reference
+   |
+   = note: expected type `Foo`
+   = note:    found type `&_`
```

Comment:
For a while I've been thinking wether it would make sense to make the `expected`/`found` notes a first class comment, so that these errors would look closer to

```
error[E0308]: mismatched types
  --> ../../src/test/ui/mismatched_types/issue-38371.rs:14:8
   |
14 | fn foo(&foo: Foo) {
   |        ^^^^ expected struct `Foo`, found reference
   |
   = expected type `Foo`
   =    found type `&_`
   = help: did you mean `foo: &Foo`?

error[E0308]: mismatched types
  --> ../../src/test/ui/mismatched_types/issue-38371.rs:26:9
   |
26 | fn agh(&&bar: &u32) {
   |         ^^^^ expected u32, found reference
   |
   = expected type `u32`
   =    found type `&_`
```

If you feel there'd be value on that, I could create an issue and start a PR for it.